--- ./gstavviddec.c	2026-03-02 09:17:16.000000000 +0100
+++ ./gstavviddec-new.c	2026-06-04 16:10:20.537686951 +0200
@@ -131,6 +131,49 @@
   return ffmpegdec_lowres_type;
 }
 
+/**
+ * GstLibAVVidDecSkipFrame:
+ *
+ * Types of frames to skip during decoding.
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::default
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::non-ref
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::bidir
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::non-intra
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::non-key
+ *
+ * Since: 1.26
+ */
+
+/**
+ * GstLibAVVidDecSkipFrame::all
+ *
+ * Since: 1.26
+ */
 #define GST_FFMPEGVIDDEC_TYPE_SKIPFRAME (gst_ffmpegviddec_skipframe_get_type())
 static GType
 gst_ffmpegviddec_skipframe_get_type (void)
@@ -139,10 +182,14 @@
 
   if (!ffmpegdec_skipframe_type) {
     static const GEnumValue ffmpegdec_skipframe[] = {
-      {0, "0", "Skip nothing"},
-      {1, "1", "Skip B-frames"},
-      {2, "2", "Skip IDCT/Dequantization"},
-      {5, "5", "Skip everything"},
+      // taken from https://ffmpeg.org
+      {AVDISCARD_DEFAULT, "Discard useless packets like 0 size packets in avi",
+          "default"},
+      {AVDISCARD_NONREF, "Discard all non reference", "non-ref"},
+      {AVDISCARD_BIDIR, "Discard all bidirectional frames", "bidir"},
+      {AVDISCARD_NONINTRA, "Discard all non intra frames", "non-intra"},
+      {AVDISCARD_NONKEY, "Discard all frames except keyframes", "non-key"},
+      {AVDISCARD_ALL, "Discard all", "all"},
       {0, NULL, NULL},
     };
 
@@ -1709,6 +1755,11 @@
   if (frame == NULL)
     return;
 
+  if (ffmpegdec->skip_frame != AVDISCARD_DEFAULT) {
+    /* A special skip frame mode is configured, ignore QOS. */
+    return;
+  }
+
   if (skip_flags & GST_SEGMENT_FLAG_TRICKMODE_KEY_UNITS) {
     ffmpegdec->context->skip_frame = AVDISCARD_NONKEY;
     *mode_switch = TRUE;
